gutenTAG/base_oscillations/custom_input.py

```diff
@@ -0,0 +1,94 @@
+    
+    def generate_only_base(self,
+                        ctx: BOGenerationContext,
+                        input_timeseries_path_test: str = None,
+                        use_column_test: Union[str, int] = None,
+                        length: Optional[int] = None,
+                        input_timeseries_path_train: Optional[str] = None,
+                        use_column_train: Optional[Union[str, int]] = None,
+                        semi_supervised: Optional[bool] = None,
+                        supervised: Optional[bool] = None,
+                        *args, **kwargs) -> np.ndarray:
+        
+        """
+             Generates a numpy array of time-series data from a CSV file based on the specified parameters.
+
+             Precondition: if specifying column with an integer, the timestamp column is at column index 0
+
+             Args:
+                 ctx (BOGenerationContext): An instance of the BOGenerationContext class.
+                 input_timeseries_path_test (str, optional): The path to the test data CSV file. Defaults to None.
+                 use_column_test (Union[str, int], optional): The name or index of the column containing the test data. 
+                     Defaults to None.
+                 length (Optional[int], optional): The desired length of the output time-series data. Defaults to None.
+                 input_timeseries_path_train (Optional[str], optional): The path to the training data CSV file. 
+                     Defaults to None.
+                 use_column_train (Optional[Union[str, int]], optional): The name or index of the column containing the training data. 
+                     Defaults to None.
+                 semi_supervised (Optional[bool], optional): A flag to indicate if the model is trained in semi-supervised mode.
+                     Defaults to None.
+                 supervised (Optional[bool], optional): A flag to indicate if the model is trained in supervised mode.
+                     Defaults to None.
+                 *args: Variable length argument list.
+                 **kwargs: Arbitrary keyword arguments.
+             
+             Returns:
+                 np.ndarray: A numpy array of the generated time-series data.
+             
+             Raises:
+                 ValueError: If the number of rows in the input timeseries file is less than the desired length.
+             
+             
+        """
+        length = length or self.length
+        input_timeseries_path_train = input_timeseries_path_train or self.input_timeseries_path_train
+        input_timeseries_path_test = input_timeseries_path_test or self.input_timeseries_path_test
+        use_column_train = use_column_train or self.use_column_train
+        use_column_test = use_column_test or self.use_column_test
+        
+        
+        if isinstance(use_column_test,str):
+            test_columns = [TIMESTAMP, use_column_test]
+            index_column_test = TIMESTAMP
```

Comment:
If you declare these variables with a union type, the type checker should be happy again:
```suggestion
            test_columns: Union[Sequence[str], Sequence[int]] = [TIMESTAMP, use_column_test]
            index_column_test: Union[str, int] = TIMESTAMP
```